tracer/src/Datadog.Trace/Telemetry/TelemetrySettings.cs

```diff
@@ -94,7 +102,18 @@ public static TelemetrySettings FromSource(IConfigurationSource? source)
                 agentless = new AgentlessSettings(agentlessUri, apiKey!);
             }
 
-            return new TelemetrySettings(telemetryEnabled, configurationError, agentless);
+            return new TelemetrySettings(telemetryEnabled, configurationError, agentless, agentProxyEnabled: agentProxyEnabled);
+        }
+
+        private static bool? IsAgentAvailable()
```

Comment:
nit - IsAgentProbablyUnavailable may better represent the content of this helper. But thiis is me and my poor naming skills :p

Reply:
😂 yeah maybe, we can easily rename it as it's internal if others feel the same 😉